solutions/src/test/java/ru/sberbank/school/task02/ReverseCalculatorTest.java

```diff
@@ -0,0 +1,125 @@
+import org.junit.rules.ExpectedException;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Optional;
+
+public class ReverseCalculatorTest {
+    private ServiceFactory factory = new ServiceFactoryImpl();
+    private ExternalQuotesService quotesService = new ExternalQuotesProvider();
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void convertReversedNoDeltaSell() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.SELL, Symbol.USD_RUB,
+                BigDecimal.valueOf(1000), Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.001).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    public void convertReversedNoDeltaBuy() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB,
+                BigDecimal.valueOf(1000), Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.005).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    public void convertReversedZeroDeltaBuy() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB,
+                BigDecimal.valueOf(1000), 0, Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.005).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    public void convertReversedDeltaBuy() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB,
+                BigDecimal.valueOf(10000), 20, Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.0025).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    public void convertReversedNullOperationThrowsNPEWithSpecificMessage() {
```

Comment:
Abbreviation in name &#39;convertReversedNullOperationThrowsNPEWithSpecificMessage&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1893531824>*